frontend/nexusInteraction.cpp

```diff
@@ -265,12 +266,28 @@ void MainWindow::handle_result_contextMenu(HttpRequestWorker *worker)
     if (worker->error_type == QNetworkReply::NoError)
     {
         contextMenu_->clear();
-        foreach (const QJsonValue &value, worker->json_array)
+
+        foreach (const QJsonValue &log, worker->json_array)
         {
-            // Fills contextMenu with all columns
-            QAction *action = new QAction(value.toString(), this);
-            connect(action, SIGNAL(triggered()), this, SLOT(contextGraph()));
-            contextMenu_->addAction(action);
+            QJsonArray logArray = log.toArray();
+            QMenu *subMenu = new QMenu(logArray.first().toString());
+            logArray.removeFirst();
+            if (logArray.size() > 0)
+                contextMenu_->addMenu(subMenu);
+
+            QVariantList logArrayVar = logArray.toVariantList();
+            std::sort(logArrayVar.begin(), logArrayVar.end(),
+                      [](QVariant &v1, QVariant &v2) { return v1.toString() < v2.toString(); });
+
+            foreach (const QVariant &block, logArrayVar)
+            {
+                // Fills contextMenu with all columns
+                QString path = block.toString();
+                QAction *action = new QAction(path.right(path.size() - path.lastIndexOf("/") - 1), this);
+                action->setData(path);
+                connect(action, SIGNAL(triggered()), this, SLOT(contextGraph()));
+                subMenu->addAction(action);
+            }
```

Comment:
Can sprinkle in some `auto`s here. There are also other places where similar stuff could be done, but I'll leave you to find those!
```suggestion
            foreach (const auto &block, logArrayVar)
            {
                // Fills contextMenu with all columns
                auto path = block.toString();
                auto *action = new QAction(path.right(path.size() - path.lastIndexOf("/") - 1), this);
                action->setData(path);
                connect(action, SIGNAL(triggered()), this, SLOT(contextGraph()));
                subMenu->addAction(action);
            }
```